<a href="https://colab.research.google.com/github/lwerth/escuelas_crimen_cdmx/blob/master/cdmx_escuelas_merged.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np, scipy as sp, pymc3 as pm, matplotlib.pyplot as plt, pandas as pd, theano, theano.tensor as tt
from scipy import stats
import seaborn as sb
from sklearn.preprocessing import OrdinalEncoder, StandardScaler

from keras.preprocessing.text import Tokenizer,  text_to_word_sequence
import re
import nltk
from nltk import tokenize
import seaborn as sns
import json

%matplotlib inline
import matplotlib.pyplot as plt
from matplotlib.pyplot import *

import theano

In [7]:
#import fidegar files

fidegar = pd.read_excel('/content/drive/My Drive/Projects/CDMX Schools/fidegar.xlsx')
fidegar_idx = OrdinalEncoder(dtype = np.int64).fit_transform(fidegar[['NIVEL']])

fidegar['nivel'] = fidegar['NIVEL']
fidegar['nombre'] = fidegar['NOMBRE']
fidegar_idx
n_niveles_fidegar = len(fidegar.nivel.unique())
n_niveles_fidegar

fidegar.nivel.unique()
fidegar

NIVEL                  NOMBRE  ...       nivel                  nombre
0       PRIMARIA        27 DE SEPTIEMBRE  ...    PRIMARIA        27 DE SEPTIEMBRE
1       PRIMARIA    ADOLFO RUIZ CORTINES  ...    PRIMARIA    ADOLFO RUIZ CORTINES
2       PRIMARIA            ALBERTO LENZ  ...    PRIMARIA            ALBERTO LENZ
3     SECUNDARIA  ALEJANDRO VON HUMBOLDT  ...  SECUNDARIA  ALEJANDRO VON HUMBOLDT
4     PREESCOLAR          ALFONSO QUIROZ  ...  PREESCOLAR          ALFONSO QUIROZ
...          ...                     ...  ...         ...                     ...
1664    PRIMARIA    VICENTE RIVA PALACIO  ...    PRIMARIA    VICENTE RIVA PALACIO
1665  SECUNDARIA              XOCHIMILCO  ...  SECUNDARIA              XOCHIMILCO
1666  PREESCOLAR             XOCHIPILOYA  ...  PREESCOLAR             XOCHIPILOYA
1667  PREESCOLAR                 XOCHITL  ...  PREESCOLAR                 XOCHITL
1668  PREESCOLAR              YOLILIZTLI  ...  PREESCOLAR              YOLILIZTLI

[1669 rows x 10 columns]

In [8]:
# read in and look at gob data

#parse text in delegacion, delete the word "delegacion" in data, alphabetize, create index, and join!

publicas = pd.read_excel('/content/drive/My Drive/Projects/CDMX Schools/escuelas-publicas.xlsx')
pub_idx = OrdinalEncoder(dtype = np.int64).fit_transform(publicas[['nivel']])

n_niveles_pub = len(publicas.nivel.unique())
n_niveles_pub

n_delegacion_pub = len(publicas.Delegacion.unique())


print(n_delegacion_pub, n_niveles_pub)
print('delegacion',publicas.Delegacion.unique())
print('niveles',publicas.nivel.unique())
publicas


36 14
delegacion ['DELEGACION IZTACALCO' 'DELEGACION COYOACAN' 'DELEGACION TLAHUAC'
 'DELEGACION IZTAPALAPA' 'DELEGACION VENUSTIANO CARRANZA'
 'DELEGACION CUAUHTEMOC' 'DELEGACION TLALPAN' 'DELEGACION MILPA ALTA'
 'DELEGACION MIGUEL HIDALGO' 'DELEGACION ALVARO OBREGON'
 'DELEGACION CUAJIMALPA DE MORELOS' 'DELEGACION XOCHIMILCO' nan
 'DELEGACION GUSTAVO A. MADERO' 'DELEGACION BENITO JUAREZ'
 'BATALLONES ROJOS NO 205 LA ALBARRADA' 'AZCAPOTZALCO'
 'SALVADOR DIAZ MIRON' 'SELENE' 'ALVARO OBREGON'
 'AV DEL TRABAJO NO 258 MORELOS' 'DELEGACION ALVARO OBREGÓN' 'SANTA ROSA'
 'CERRADAS JUAN ESCUTIA' 'MEXICO DF.' 'PRIMARIA GENERAL'
 'DELEGACION AZCAPOTZALCO OBREGON' 'ADOLFO LOPEZ MATEOS' 'EL ROSARIO'
 'TELESECUNDARIA 89' 'EMILIO CARRANZA NO 39 ALBERT.' 'SAN JOSE'
 'SAN JUAN DE ARAGON' '553 Y 504 SECC. VECINAL DOS' '-UNIDAD HA'
 'SANTA CRUZ MEYEHUALCO']
niveles ['INTERVENCION' 'PRIMARIA' 'PREESCOLAR' 'ESCUELA' 'SECUNDARIA'
 'TELESECUNDARIA' 'LACTANTE,' 'SEC.' 'BRINDAR' 'SUPERVISAR' 'ATENDER'
 'LICE

nivel    nivel_2      nivel_3  ...      Ciudad Unnamed: 12 Unnamed: 13
0     INTERVENCION  TEMPRANA,  PREESCOLAR,  ...  MEXICO DF.         NaN         NaN
1         PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2       PREESCOLAR    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
3       PREESCOLAR    GENERAL          CON  ...  MEXICO DF.         NaN         NaN
4         PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
...            ...        ...          ...  ...         ...         ...         ...
2378      PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2379    SECUNDARIA    GENERAL          NaN  ...         NaN         NaN         NaN
2380    PREESCOLAR    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2381    SECUNDARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2382      PRIMARIA    GENERAL          NaN  ...         NaN         NaN         NaN

[2383 rows x 14 columns]

In [9]:
niveles_conjun = pd.merge(publicas, fidegar)
# niveles_conjun

## Index the nombres of each dataset

#### Merege and drop duplicates
Here we end up multiplying the datasets to get 1840 rows, adding about 200 to their count and dropping 500 from ours. The original fidegar has 1669 and the origianl escuelas publicas has 2383, i.e. fidegar has ~714 less schools listed.

<!-- I dropped the duplicates and still got 3049 rows, this seems promising.  -->

> *I'm going to keep this set without duplicates and seet how that helps us.*

In [10]:
#cerate index of nombre and merge there
pub_nombre_idx = publicas.set_index('nombre')
fide_nombre_idx = fidegar.set_index('nombre')
# display(pub_nivel_idx, fide_nivel_idx)

conjun_nombre_idx = pd.merge(pub_nombre_idx, fide_nombre_idx,
                      left_index = True, right_index = True)
conjun_nombre_idx

nivel_x  nivel_2  ...           ALCALDÍA     nivel_y
nombre                                ...                               
14 DE JULIO        PRIMARIA  GENERAL  ...       AZCAPOTZALCO    PRIMARIA
20 DE NOVIEMBRE  PREESCOLAR  GENERAL  ...  GUSTAVO A. MADERO  PREESCOLAR
21 DE ABRIL      PREESCOLAR  GENERAL  ...           COYOACAN  PREESCOLAR
21 DE MARZO        PRIMARIA  GENERAL  ...         CUAUHTEMOC    PRIMARIA
24 DE FEBRERO      PRIMARIA  GENERAL  ...         IZTAPALAPA    PRIMARIA
...                     ...      ...  ...                ...         ...
YOLOXOCHITL      PREESCOLAR  GENERAL  ...     ALVARO OBREGON  PREESCOLAR
YURI A. GAGARIN  SECUNDARIA  GENERAL  ...            TLALPAN  SECUNDARIA
ZACATENCO        PREESCOLAR  GENERAL  ...  GUSTAVO A. MADERO  PREESCOLAR
ZIMBABWE           PRIMARIA  GENERAL  ...     ALVARO OBREGON    PRIMARIA
ZOFAI            PREESCOLAR  GENERAL  ...         IZTAPALAPA  PREESCOLAR

[1878 rows x 22 columns]

In [18]:
conjun_nombre_idx.to_excel(r'C:\Users\LW\Desktop\cdmx-escuelas_merged_2020.xlsx', index = False, header=True)
# conjun_nombre_idx.to_csv(r'C:\Users\LW\Documents\Project Research\cdmx-escuelas_merged.csv', index = False, header=True)

# conjun_nombre_idx.to_excel("C:\\Users\\LW\\Documents\\Project Research\\escuelas.xlsx")  

# with pd.ExcelWriter('cdmx-escuelas_merged.xlsx',
#                     mode='a') as writer:  
#     conjun_nombre_idx.to_excel(writer, sheet_name='conjun_nombre_idx')
    
    # conjun_nombre_idx.to_excel('cdmx-escuelas_merged.csv')


In [11]:
# conjun_nombre_idx = conjun_nombre_idx.drop(columns=['NIVEL', 'NOMBRE', 'CCT 1', 'CCT 2', 'CCT 3', 'CCT 4', 'MATRÍCULA', 'ALCALDÍA', 'ESCUELAS EN PROCESO JURÍDICO', 'nombre_y'])
conjun_nombre_idx = conjun_nombre_idx.drop_duplicates()
conjun_nombre_idx

nivel_x  nivel_2  ...           ALCALDÍA     nivel_y
nombre                                ...                               
14 DE JULIO        PRIMARIA  GENERAL  ...       AZCAPOTZALCO    PRIMARIA
20 DE NOVIEMBRE  PREESCOLAR  GENERAL  ...  GUSTAVO A. MADERO  PREESCOLAR
21 DE ABRIL      PREESCOLAR  GENERAL  ...           COYOACAN  PREESCOLAR
21 DE MARZO        PRIMARIA  GENERAL  ...         CUAUHTEMOC    PRIMARIA
24 DE FEBRERO      PRIMARIA  GENERAL  ...         IZTAPALAPA    PRIMARIA
...                     ...      ...  ...                ...         ...
YOLOXOCHITL      PREESCOLAR  GENERAL  ...     ALVARO OBREGON  PREESCOLAR
YURI A. GAGARIN  SECUNDARIA  GENERAL  ...            TLALPAN  SECUNDARIA
ZACATENCO        PREESCOLAR  GENERAL  ...  GUSTAVO A. MADERO  PREESCOLAR
ZIMBABWE           PRIMARIA  GENERAL  ...     ALVARO OBREGON    PRIMARIA
ZOFAI            PREESCOLAR  GENERAL  ...         IZTAPALAPA  PREESCOLAR

[1878 rows x 22 columns]

## Index the niveles of each dataset

#### Merege and drop duplicates
Here we end up multiplying the datasets to get ~2220000 rows, then I dropped the duplicates and got only 2140 rows. The original fidegar has 2696 and the origianl escuelas publicas has 2383, i.e. fidegar has ~300 more schools listed.

In [12]:
#cerate index of nivel and merge there
pub_nivel_idx = publicas.set_index('nivel')
fide_nivel_idx = fidegar.set_index('nivel')
# display(pub_nivel_idx, fide_nivel_idx)

conjun_idx = pd.merge(pub_nivel_idx, fide_nivel_idx,
                      left_index = True, right_index = True)
conjun_idx


nivel_2 nivel_3  ...        ALCALDÍA                     nombre_y
nivel                        ...                                             
PREESCOLAR  GENERAL     NaN  ...  ALVARO OBREGON               ALFONSO QUIROZ
PREESCOLAR  GENERAL     NaN  ...  ALVARO OBREGON                    ANA FRANK
PREESCOLAR  GENERAL     NaN  ...  ALVARO OBREGON              ANTONIO VIVALDI
PREESCOLAR  GENERAL     NaN  ...  ALVARO OBREGON          BARTOLOME DE MEDINA
PREESCOLAR  GENERAL     NaN  ...  ALVARO OBREGON         CENTRO CULTURAL ALFA
...             ...     ...  ...             ...                          ...
SECUNDARIA  GENERAL     NaN  ...      XOCHIMILCO  ROSARIO GUTIERREZ ESKILDSEN
SECUNDARIA  GENERAL     NaN  ...      XOCHIMILCO           SECUNDARIA FEDERAL
SECUNDARIA  GENERAL     NaN  ...      XOCHIMILCO           TELESECUNDARIA 121
SECUNDARIA  GENERAL     NaN  ...      XOCHIMILCO              TLAMACHTILCALLI
SECUNDARIA  GENERAL     NaN  ...      XOCHIMILCO                   XOCHIMILCO

[1388433 rows x 22 columns]

In [17]:
conjun_idx = conjun_idx.drop(columns=['CCT 1', 'CCT 2', 'CCT 3', 'CCT 4', 'MATRÍCULA', 'ALCALDÍA', 'Unnamed: 12', 'Unnamed: 13', 'nombre_y'])
conjun_idx = conjun_idx.drop_duplicates()
conjun_idx

nivel_2 nivel_3  ...       NIVEL                       NOMBRE
nivel                        ...                                         
PREESCOLAR  GENERAL     NaN  ...  PREESCOLAR               ALFONSO QUIROZ
PREESCOLAR  GENERAL     NaN  ...  PREESCOLAR                    ANA FRANK
PREESCOLAR  GENERAL     NaN  ...  PREESCOLAR              ANTONIO VIVALDI
PREESCOLAR  GENERAL     NaN  ...  PREESCOLAR          BARTOLOME DE MEDINA
PREESCOLAR  GENERAL     NaN  ...  PREESCOLAR         CENTRO CULTURAL ALFA
...             ...     ...  ...         ...                          ...
SECUNDARIA  GENERAL     NaN  ...  SECUNDARIA                 JUAN ALVAREZ
SECUNDARIA  GENERAL     NaN  ...  SECUNDARIA  ROSARIO GUTIERREZ ESKILDSEN
SECUNDARIA  GENERAL     NaN  ...  SECUNDARIA           TELESECUNDARIA 121
SECUNDARIA  GENERAL     NaN  ...  SECUNDARIA              TLAMACHTILCALLI
SECUNDARIA  GENERAL     NaN  ...  SECUNDARIA                   XOCHIMILCO

[1329229 rows x 13 columns]

## Drop words to merge easier

here I try to play around with the stings in the columns to see if we could get matches ... it didn't exactly work.

In [ ]:
publicas['Delegacion'] = publicas['Delegacion'].str.strip('DELEGACION')

publicas

nivel    nivel_2      nivel_3  ...      Ciudad Unnamed: 12 Unnamed: 13
0     INTERVENCION  TEMPRANA,  PREESCOLAR,  ...  MEXICO DF.         NaN         NaN
1         PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2       PREESCOLAR    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
3       PREESCOLAR    GENERAL          CON  ...  MEXICO DF.         NaN         NaN
4         PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
...            ...        ...          ...  ...         ...         ...         ...
2378      PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2379    SECUNDARIA    GENERAL          NaN  ...         NaN         NaN         NaN
2380    PREESCOLAR    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2381    SECUNDARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2382      PRIMARIA    GENERAL          NaN  ...         NaN         NaN         NaN

[2383 rows x 14 columns]

In [ ]:
def clean_text(text):
  text = text.remove(r"DELEGACION", r'', text)
  text = ' '.join(text.split())
  return text

publicas['delegaciones'] = publicas['Delegacion'].apply(lambda x: clean_text(x))


In [ ]:
drop_wrd = ['DELEGACION']
strings = publicas.Delegacion[:]
strings = ''.join((filter(lambda i: i not in drop_wrd, strings)))
print(strings)

TypeError: ignored

In [ ]:
publicas.delegaciones = []

for i in publicas.Delegacion:
  delegaciones = str(publicas.Delegacion)
  words = delegaciones.split()
  if i in words == "DELEGACION":
    drop = words.remove('DELEGACION')
    redo = ''.join(drop)
    publicas.delegaciones.append(redo)

publicas


nivel    nivel_2      nivel_3  ...      Ciudad Unnamed: 12 Unnamed: 13
0     INTERVENCION  TEMPRANA,  PREESCOLAR,  ...  MEXICO DF.         NaN         NaN
1         PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2       PREESCOLAR    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
3       PREESCOLAR    GENERAL          CON  ...  MEXICO DF.         NaN         NaN
4         PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
...            ...        ...          ...  ...         ...         ...         ...
2378      PRIMARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2379    SECUNDARIA    GENERAL          NaN  ...         NaN         NaN         NaN
2380    PREESCOLAR    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2381    SECUNDARIA    GENERAL          NaN  ...  MEXICO DF.         NaN         NaN
2382      PRIMARIA    GENERAL          NaN  ...         NaN         NaN         NaN

[2383 rows x 14 columns]